cmd/lifecycle/platform/analyze_inputs.go

```diff
@@ -0,0 +1,124 @@
+package platform
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/lifecycle"
+	"github.com/buildpacks/lifecycle/api"
+	"github.com/buildpacks/lifecycle/image"
+	"github.com/buildpacks/lifecycle/internal/str"
+)
+
+// AnalyzeInputs holds the values of command-line flags and args.
+// Fields are the cumulative total of inputs across all supported platform APIs.
+type AnalyzeInputs struct { // TODO: make "to AnalyzerOpts" method? Do these structs need to be different?
```

Comment:
I wonder if it might read better to do an options pattern for these type of structs. Just thinking out loud, it might not make sense.

```go
inputs, err := NewAnalyzeInputs(platformAPI,  logger, 
  WithCacheImage(),
  WithLayersDir(),
  ...
)
```